src/System.Management.Automation/engine/remoting/client/remoterunspace.cs

```diff
@@ -2005,27 +2005,213 @@ public override void StopProcessCommand()
+                {
+                    { "Id", id },
+                });
+        }
 
         /// <summary>
         /// Returns breakpoints primarily for the Get-PSBreakpoint cmdlet.
         /// </summary>
-        public override List<Breakpoint> GetBreakpoints() =>
-            _runspace.Debugger?.GetBreakpoints();
+        public override List<Breakpoint> GetBreakpoints()
+        {
+            // This is supported only for PowerShell versions >= 7.0
+            if (!_remoteDebuggingCapability.IsCommandSupported(RemoteDebuggingCommands.GetBreakpoint))
+            {
+                throw new PSNotSupportedException(
+                    StringUtil.Format(
+                        DebuggerStrings.CommandNotSupportedForRemoteUseInServerDebugger,
+                        RemoteDebuggingCommands.CleanCommandName(RemoteDebuggingCommands.GetBreakpoint)));
+            }
+
+            CheckForValidateState();
+
+            var breakpoints = new List<Breakpoint>();
+
+            using (PowerShell ps = GetNestedPowerShell())
+            {
+                ps.AddCommand(RemoteDebuggingCommands.GetBreakpoint);
+
+                Collection<PSObject> output = ps.Invoke<PSObject>();
+                foreach (var item in output)
+                {
+                    if (item?.BaseObject is Breakpoint)
+                    {
+                        breakpoints.Add((Breakpoint)item.BaseObject);
+                    }
+                }
+            }
+
+            return breakpoints;
+        }
+
+        public override CommandBreakpoint SetCommandBreakpoint(string command, ScriptBlock action = null, string path = null)
+        {
+            // This is supported only for PowerShell versions >= 7.0
+            if (!_remoteDebuggingCapability.IsCommandSupported(RemoteDebuggingCommands.SetBreakpoint))
+            {
+                throw new PSNotSupportedException(
+                    StringUtil.Format(
+                        DebuggerStrings.CommandNotSupportedForRemoteUseInServerDebugger,
+                        RemoteDebuggingCommands.CleanCommandName(RemoteDebuggingCommands.SetBreakpoint)));
+            }
+
+            var functionParameters = new Dictionary<string, object>
+            {
+                { "Command", command },
+            };
+
+            if (action != null)
+            {
+                functionParameters.Add("Action", action);
+            }
+
+            if (path != null)
+            {
+                functionParameters.Add("Script", path);
+            }
+
+            return InvokeRemoteBreakpointFunction<CommandBreakpoint>(RemoteDebuggingCommands.SetBreakpoint, functionParameters);
+        }
+
+        public override LineBreakpoint SetLineBreakpoint(string path, int line, int column = 0, ScriptBlock action = null)
+        {
+            // This is supported only for PowerShell versions >= 7.0
+            if (!_remoteDebuggingCapability.IsCommandSupported(RemoteDebuggingCommands.SetBreakpoint))
```

Comment:
We should consider caching this value, since this does not change. Also, it is used from 3 APIs.

Reply:
Good idea. I cache the check across all debugger APIs since they're all enabled in the same release.